packages/core/lib/util/calculate-byte-length.ts

```diff
@@ -0,0 +1,17 @@
+// returns the byte length of an utf8 string
+export function calculateByteLength(str: string): number {
```

Comment:
can just be `TextEncoder` in browsers/workers and from `util` on node. No reason to include this.

Reply:
We benchmarked it and it was so slow on Node.js

Reply:
So the web/browser has to pay the price? chuck it in a preamble or bundle for the different targets.

Reply:
Actually encoding entire string with TextEncoder sounds more expensive, no?

Reply:
One would have to benchmark, but in reality who cares about the byte length in the first place? Its a header that nobody is looking at.

Based on my other point in this thread, why does even need to exist in the first instance.

Reply:
I think nobody refers to the HTTP Client here. As I already said below, some environments return 411 in case of missing length. Also this calculation was already in Helix before so this PR doesn't cover that actually.